l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py

```diff
@@ -617,6 +617,50 @@ def setDettaglioLinee(self, invoice, body):
                 line_no, line, body, price_precision, uom_precision)
             line_no += 1
 
+        generic_mngt_line = False
+        generic_mngt_lines = invoice.related_mngt_data_ids.filtered(
+            lambda x: not x.lineRef)
+        if generic_mngt_lines:
+            generic_mngt_line = generic_mngt_lines[0]
+        for DettaglioLinea in body.DatiBeniServizi.DettaglioLinee:
+            # get related mgnt_line if exists
+            mngt_lines = filter(
+                lambda x: x.lineRef == DettaglioLinea.NumeroLinea,
+                invoice.related_mngt_data_ids)
+            if mngt_lines:
+                for mngt_line in mngt_lines:
+                    dati_gestionali = AltriDatiGestionaliType()
+                    if mngt_line.name:
+                        dati_gestionali.TipoDato = mngt_line.name
+                    if mngt_line.text_ref:
+                        dati_gestionali.RiferimentoTesto = mngt_line.text_ref
+                    if mngt_line.number_ref:
+                        dati_gestionali.RiferimentoNumero = '%.2f' % \
+                            mngt_line.number_ref
+                    if mngt_line.date_ref:
+                        dati_gestionali.RiferimentoData = mngt_line.date_ref
+                    DettaglioLinea.AltriDatiGestionali.append(
+                        dati_gestionali
+                    )
+            else:
+                if generic_mngt_line:
+                    dati_gestionali = AltriDatiGestionaliType()
+                    # if fatturapa line is not referred, and exist a
+                    # generic_mngt_line, add this generic mngt data to line
+                    if generic_mngt_line.name:
```

Comment:
Questo codice è duplicato di quello sopra, sarebbe bene estrarre un metodo da usare in entrambi i casi